tower-grpc-build/src/lib.rs

```diff
@@ -101,46 +84,40 @@ impl Config {
 }
 
 impl prost_build::ServiceGenerator for ServiceGenerator {
-    fn generate(&self, service: prost_build::Service, buf: &mut String) {
-        let inner = self.inner.borrow();
-        let mut state = self.state.borrow_mut();
-
-        // if a service was already generated, remove its code from the 
-        // `prost_build` buffer, because it's already in the `codegen::Scope`, 
-        // and we will add to existing modules in the generated code. we can't
-        // just wipe out the whole buffer, because the code generated by 
-        // `prost` is in there and we need that.
-        // XXX rewrite to use `ServiceGenerator::finalize` from `prost-build`.
-        if state.last_gen_length > 0 {
-            let new_len = buf.len() - state.last_gen_length;
-            buf.truncate(new_len);
-        }
 
-        // Add an extra new line to separate messages
-        write!(buf, "\n").unwrap();
+    fn generate(&self, service: prost_build::Service, _buf: &mut String) {
```

Comment:
`prost` could probably further simplify this by making `ServiceGenerator::generate` and `ServiceGenerator::finalize` take `&mut self`.  Then you wouldn't have to mess with refcell.  @vorner what do you think?

Reply:
I thought about that and decided it would be better, but it would also be a backwards-incompatible change, so I stayed with the original shared borrow in the pull request.

Reply:
Switching to `&mut self` would be better for us :)